concrete/src/Database/EntityManager/Provider/AbstractPackageProvider.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace Concrete\Core\Database\EntityManager\Provider;
+
+use Concrete\Core\Application\Application;
+use Concrete\Core\Database\EntityManager\Provider\ProviderInterface;
+use Concrete\Core\Package\Package;
+use Doctrine\ORM\Mapping\Driver\AnnotationDriver;
+
+abstract class AbstractPackageProvider implements ProviderInterface
+{
+
+    /**
+     * @var Package
+     */
+    protected $pkg;
+    protected $app;
+
+    public function __construct(Package $pkg)
+    {
+        $this->app = $pkg->getApplication();
```

Comment:
This might be better off as a hard dependency IE:

```php
__construct(Package $pkg, Application $app);
```

Ideally the package object won't be a source of truth for the application object